rs/replay/src/player.rs

```
@@ -1294,13 +1255,89 @@ impl Player {
+    version: u64,
+    ingress_expiry: Time,
+    runtime: &Runtime,
+    perform_query: &(impl PerformQuery + Sync),
+) -> Result<Vec<RegistryTransportRecord>, String> {
+    let payload = serialize_get_changes_since_request(version).unwrap();
+    let query = Query {
+        source: QuerySource::User {
+            user_id: UserId::from(PrincipalId::new_anonymous()),
+            ingress_expiry: ingress_expiry.as_nanos_since_unix_epoch(),
+            nonce: None,
+        },
+        receiver: REGISTRY_CANISTER_ID,
+        method_name: "get_changes_since".to_string(),
+        method_payload: payload,
+    };
+    match runtime
+        .block_on(perform_query.perform_query(query))
+        .unwrap()
+    {
+        Ok((Ok(wasm_result), _time)) => match wasm_result {
+            WasmResult::Reply(v) => {
+                let (high_capacity_deltas, _version) = deserialize_get_changes_since_response(v)
+                    .map_err(|err| format!("{:?}", err))?;
+
+                // Dechunkify deltas.
+                let mut inlined_deltas = vec![];
+                for delta in high_capacity_deltas {
+                    let get_chunk = GetChunkImpl {
+                        perform_query,
+                        ingress_expiry,
+                    };
+
+                    let delta = runtime
+                        .block_on(dechunkify_delta(delta, &get_chunk))
+                        .map_err(|err| format!("{:?}", err))?;
+
+                    inlined_deltas.push(delta);
+                }
+
+                registry_deltas_to_registry_transport_records(inlined_deltas)
+                    .map_err(|err| format!("{:?}", err))
+            }
+
+            WasmResult::Reject(e) => Err(format!("Query rejected: {}", e)),
+        },
+        Ok((Err(err), _)) => Err(format!("Query failed: {:?}", err)),
+        Err(QueryExecutionError::CertifiedStateUnavailable) => {
+            panic!("Certified state unavailable for query call.")
```

Seems like we could return an error instead

---

Done.

Agreed. The only reason I did it this way was that I tried to preserve the old behavior as much as possible, but I do not mind the behavior you suggest.